### DISCUSSION TITLE
Sonoma 14.5 and Xcode version 15.4 Builds Supported?

### DISCUSSION THREAD

**Discussion by ppridge** [jasonacox/Build-OpenSSL-cURL#74] [General]
I sort of got forced into upgrading to Sonoma 14.5 and that brought in XCode 15.4.  Just wondering if anyone has successfully built on this combination?   I got errors in building NGHTTT2.  I don't have the details at the moment, I can get them if there is interest in my issue.   Mostly, I just want to know if a others have been successful building this combination.

Thanks much,  
tom

**Comment by jasonacox** [comment]
Hi @ppridge - I have a M2 on Sonoma 14.5 with Xcode 15.3 that builds fine. I'll try upgrading Xcode. Can you paste the errors you see?

**Reply by ppridge** [reply]
Thank you, I will do so this afternoon.

**Comment by ppridge** [comment]
Jason,

Here is the System, Machine, and Dev Tooll Info as well as the build logs.  Please let me know if you need any addition information or wish me to try or install any other package versions.  Thanks again for looking into this.  -- tom

OPENSSL build.sh Failure

system_profiler SPSoftwareDataType
Software:
    System Software Overview:
      System Version: macOS 14.5 (23F79)
      Kernel Version: Darwin 23.5.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 3 days, 20 hours, 31 minutes

system_profiler SPHardwareDataType
Hardware:
    Hardware Overview:
      Model Name: Mac mini
      Model Identifier: Macmini9,1
      Model Number: Z12N000G0LL/A
      Chip: Apple M1
      Total Number of Cores: 8 (4 performance and 4 efficiency)
      Memory: 16 GB
      System Firmware Version: 10151.121.1
      OS Loader Version: 10151.121.1
      Activation Lock Status: Disabled

system_profiler SPDeveloperToolsDataType
Developer:
    Developer Tools:
      Version: 15.4 (15F31d)
      Location: /Applications/Xcode.app
      Applications:
          Xcode: 15.4 (22622)
          Instruments: 15.3 (64565.112)
      SDKs:
          DriverKit:
              23.5: 
          iOS:
              17.5: (21F77)
          iOS Simulator:
              17.5: (21F77)
          macOS:
              14.5: (23F73)
          tvOS:
              17.5: (21L566)
          tvOS Simulator:
              17.5: (21L566)
          visionOS:
              1.2: (21O5565d)
          visionOS Simulator:
              1.2: (21O5565d)
          watchOS:
              10.5: (21T569)
          watchOS Simulator:
              10.5: (21T569)

Build.sh Log
Using openssl-3.0.13.tar.gz
** Building OpenSSL openssl-3.0.13 **
Unpacking openssl
Building Mac libraries
Building openssl-3.0.13 for x86_64 (MacOS 14.5)
Building openssl-3.0.13 for arm64 (MacOS 14.5)
Testing binary for arm64:
OpenSSL 3.0.13 30 Jan 2024 (Library: OpenSSL 3.0.13 30 Jan 2024)
  Copying headers and libraries
Building tvOS libraries
Building openssl-3.0.13 for AppleTVOS  arm64 (tvOS 9.0)
  Copying headers and libraries
Building tvOS simulator libraries
 Building openssl-3.0.13 for AppleTVSimulator  arm64 (tvOS Simulator 9.0)
 Building openssl-3.0.13 for AppleTVSimulator  x86_64 (tvOS Simulator 9.0)
  Copying headers and libraries
Using openssl-3.0.13.tar.gz
** Building OpenSSL openssl-3.0.13 **
Unpacking openssl
Building iOS libraries
Building openssl-3.0.13 for iPhoneOS  armv7 (iOS 8.0)
Building openssl-3.0.13 for iPhoneOS  armv7s (iOS 8.0)
Building openssl-3.0.13 for iPhoneOS  arm64 (iOS 8.0)
Building openssl-3.0.13 for iPhoneOS  arm64e (iOS 8.0)
Building openssl-3.0.13 for iPhoneSimulator  i386 (iOS 8.0)
Building openssl-3.0.13 for iPhoneSimulator  x86_64 (iOS 8.0)
Building openssl-3.0.13 for iPhoneSimulator  arm64 (iOS 8.0)
  Copying headers and libraries
  Creating combined OpenSSL libraries for iOS
Cleaning up
Done

** WARNING: pkg-config not installed... attempting to install.
  Downloading pkg-config-0.29.2.tar.gz
** ERROR with Build - Check /tmp/nghttp2*.log
make[4]: *** [all] Error 2
make[3]: *** [all-recursive] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1

make: *** [all] Error 2

 


nghttp2-1.60.0.log 

 CC       libglib_2_0_la-gatomic.lo
gatomic.c:392:10: error: incompatible integer to pointer conversion passing 'gssize' (aka 'long') to parameter of type 'gpointer' (aka 'void *') [-Wint-conversion]
  return g_atomic_pointer_add ((volatile gpointer *) atomic, val);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./gatomic.h:170:46: note: expanded from macro 'g_atomic_pointer_add'
    (gssize) __sync_fetch_and_add ((atomic), (val));                         \
                                             ^~~~~
gatomic.c:416:10: error: incompatible integer to pointer conversion passing 'gsize' (aka 'unsigned long') to parameter of type 'gpointer' (aka 'void *') [-Wint-conversion]
  return g_atomic_pointer_and ((volatile gpointer *) atomic, val);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./gatomic.h:177:45: note: expanded from macro 'g_atomic_pointer_and'
    (gsize) __sync_fetch_and_and ((atomic), (val));                         \                                           ^~~~~

gatomic.c:440:10: error: incompatible integer to pointer conversion passing 'gsize' (aka 'unsigned long') to parameter of type 'gpointer' (aka 'void *') [-Wint-conversion]
  return g_atomic_pointer_or ((volatile gpointer *) atomic, val);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./gatomic.h:184:44: note: expanded from macro 'g_atomic_pointer_or'
    (gsize) __sync_fetch_and_or ((atomic), (val));                          \
                                           ^~~~~
gatomic.c:464:10: error: incompatible integer to pointer conversion passing 'gsize' (aka 'unsigned long') to parameter of type 'gpointer' (aka 'void *') [-Wint-conversion]
  return g_atomic_pointer_xor ((volatile gpointer *) atomic, val);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./gatomic.h:191:45: note: expanded from macro 'g_atomic_pointer_xor'
    (gsize) __sync_fetch_and_xor ((atomic), (val));                         \
                                            ^~~~~
4 errors generated.

**Reply by jasonacox** [reply]
Thanks @ppridge - I'm not able to reproduce. I upgraded to `Xcode 15.4 - Build version 15F31d` but it builds correctly without errors.  Are you specifying anything but `build.sh`?.

> ** WARNING: pkg-config not installed... attempting to install.
Downloading pkg-config-0.29.2.tar.gz

I'm wondering if this is the issue. Can you run:

```bash
$ pkg-config --version 
0.29.2
```

**Comment by ppridge** [comment]
Spot on.  You were exactly correct: pkg-config was the issue.   You mark this discussion as successfully resolved.   Thank you

**Reply by jasonacox** [reply]
How did you fix it? We may want to adjust the build.sh to help.
